### PR TITLE
Refactor Tooltip component

### PR DESCRIPTION
**Pull request by torztomasz**
This pull request refactors the Tooltip component and unifies tooltip display logic.

### CODE REVIEW

File: packages/frontend/src/components/tooltip/TooltipProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  withAnimation = true,
+}: TooltipProviderProps) {
+  return (
+    <div
+      className={classNames(
+        'fixed top-0 left-0 z-110 hidden max-w-[300px] rounded-md bg-white px-4 py-3 text-left text-sm leading-tight text-gray-700 shadow-[0px_4px_12px_0px_rgba(0,0,0,0.55)] dark:bg-neutral-700 dark:text-white',
+        withAnimation && 'animate-quick-fade-in',
+      )}
+      data-role="tooltip-popup"
+      data-testid="tooltip"
```
Comment (maciekzygmunt) [inline packages/frontend/src/components/tooltip/TooltipProvider.tsx:18]: Why do we need this?

File: packages/frontend/src/components/tooltip/Tooltip.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import React from 'react'
+
+interface TooltipProps {
+  children: React.ReactNode
```
Comment (maciekzygmunt) [inline packages/frontend/src/components/tooltip/Tooltip.tsx:4]: Maybe we can change this type to the below one, to always require usage of both components
```suggestion
  children: [
    ReactElement<TooltipTriggerProps>,
    ReactElement<TooltipContentProps>,
  ]
```

File: packages/frontend/src/components/tvl-breakdown/table/TokenAmountCell.tsx [hunk 3]
```diff
@@ -23,28 +24,30 @@ export function TokenAmountCell(props: TokenAmountCellProps) {
...
+              key={escrow.escrowAddress.toString()}
+              className="MultipleEscrowsHidden hidden font-normal text-black/80 dark:text-white/80"
+              data-token={props.assetId}
+            >
+              {formatLargeNumberWithCommas(Number(escrow.amount))}
+            </div>
+          ))}
+      </TooltipTrigger>
+      <TooltipContent>Circulating supply</TooltipContent>
+    </Tooltip>
```
Comment (maciekzygmunt) [inline packages/frontend/src/components/tvl-breakdown/table/TokenAmountCell.tsx:42]: Something is broken here:
image

File: packages/frontend/src/components/tvl-breakdown/TvlBreakdownSummaryBox.tsx [hunk 4]
```diff
@@ -75,9 +76,9 @@ function StatsItem(props: StatsItemProps) {
           {props.mobileTitle}
         </span>
 
-        <span
-          className="Tooltip ml-0.5 -translate-y-px md:translate-y-0"
-          title={props.tooltip}
+        <Tooltip
+          className="ml-0.5 -translate-y-px md:translate-y-0"
+          content={props.tooltip}
```
Comment (maciekzygmunt) [inline packages/frontend/src/components/tvl-breakdown/TvlBreakdownSummaryBox.tsx:81]: What is this?

File: packages/frontend/src/components/tooltip/TooltipProvider.tsx (hunk 1 above)
Comment (torztomasz) [reply packages/frontend/src/components/tooltip/TooltipProvider.tsx:18]: for the storybook tests, we use them in some cases

File: packages/frontend/src/components/tooltip/Tooltip.tsx (hunk 2 above)
Comment (torztomasz) [reply packages/frontend/src/components/tooltip/Tooltip.tsx:4]: goood stuuuuf

**Review by maciekzygmunt — APPROVED** [review]

**Pull request merged**